### PR TITLE
feat: Add solver HTTP tracing

### LINKED ISSUE #387: Add solver HTTP traces
We would like to collect more information about HTTP requests made to the solver server. This data can give us information about frequency of requests, status codes, failed requests and more.

Let's see if we can add https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/github.com/gorilla/mux/otelmux as a middleware on our router. We can send traces to our existing observability infrastructure.

### PR DESCRIPTION
**Pull request by bgins**
### Summary

This pull request makes the following changes:

- [x] Add solver telemetry config and tracer
- [x] Add `gorilla/mux/otelmux` middleware to solver HTTP router

The pull request adds OpenTelemetry autoinstrumentation to the solver's HTTP router. It uses our existing tracer provider to send traces to our existing observability infrastructure.

### Task/Issue reference

Closes: #387 
### Test plan

Start the observability stack (https://github.com/Lilypad-Tech/observability) with:

```
./stack observe-up dev
```

Start the stack and run a job. Open the observability dashboard at `localhost:3000` and run a TraceQL query like:

```
{resource.service.name="solver" && span.http.method="GET"}
```

### CODE REVIEW

File: cmd/lilypad/solver.go [hunk 1]
```diff
@@ -34,12 +35,18 @@ func newSolverCmd() *cobra.Command {
...
-func runSolver(cmd *cobra.Command, options solver.SolverOptions) error {
+func runSolver(cmd *cobra.Command, options solver.SolverOptions, network string) error {
 	commandCtx := system.NewCommandContext(cmd)
 	defer commandCtx.Cleanup()
 
-	noopTracer := noop.NewTracerProvider().Tracer(system.GetOTelServiceName(system.SolverService))
-	web3SDK, err := web3.NewContractSDK(commandCtx.Ctx, options.Web3, noopTracer)
+	telemetry, err := configureTelemetry(commandCtx.Ctx, system.SolverService, network, options.Telemetry, options.Web3)
+	if err != nil {
+		fmt.Printf("failed to setup opentelemetry: %s", err)
```
Comment (walkah) [inline cmd/lilypad/solver.go]: why aren't we using `log` here?

**Review by walkah — APPROVED** [review]
LGTM! Tested locally and I see HTTP traces from the solver ✅

**Review by narbs91 — APPROVED** [review]

File: cmd/lilypad/solver.go (hunk 1 above)
Comment (narbs91) [reply cmd/lilypad/solver.go]: Same nit from me too but otherwise looks good

Comment (bgins) [reply cmd/lilypad/solver.go]: No good reason for it. Updated it to a warn log and also updated similar in the resource provider:

https://github.com/Lilypad-Tech/lilypad/blob/224ee5bf4fd59b6fae638973b479778ef0f67612/cmd/lilypad/resource-provider.go#L43

**Pull request merged**